### PR TITLE
Use a global script in Tiberian Dawn 

### LINKED ISSUE #9456: Lua: standardize naming of player?
Currently, for TD missions the player is either referenced as "GDI/Nod" or "player"
- `player = Player.GetPlayer("GDI")` (gdi01)
- `gdi = Player.GetPlayer("GDI")` (gdi04a)

should this be made consistent throughout all TD missions?

I'd prefer to always use "player/enemy" for factions for two reasons:
- makes code more readable
- makes code more reusable (copy&paste a function from a gdi to a nod mission without having to deal with gdi/nod mixups; also half the missions capitalize the player as "Nod", the other half as "nod")

this should reduces the chance for bugs (uppercase VS lowercase "GDI", and accidentally referencing the wrong player)
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26982294-lua-standardize-naming-of-player?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #16833: Mission lua typo
The `GuardBase` function has a typo (Line 91) in the following mission-AI.lua:

- [nod08a-AI.lua](https://github.com/OpenRA/OpenRA/blob/2c6c2c00b40110ffeb1bbaf9cc870e7b67f0ea8d/mods/cnc/maps/nod08a/nod08a-AI.lua)
- [nod08b-AI.lua](https://github.com/OpenRA/OpenRA/blob/2c6c2c00b40110ffeb1bbaf9cc870e7b67f0ea8d/mods/cnc/maps/nod08b/nod08b-AI.lua)
- [nod09-AI.lua](https://github.com/OpenRA/OpenRA/blob/2c6c2c00b40110ffeb1bbaf9cc870e7b67f0ea8d/mods/cnc/maps/nod09/nod09-AI.lua)

`Trigger.OnDamaged(building, function(guard)`
should be
`Trigger.OnDamaged(building, function()`

**EDIT**:
Also, after Line 128 and 153 (`Trigger.AfterDelay([...])` of Functions
`ProduceInfantry = function(building)`
and
`ProduceVehicle = function(building)`
a `return` Statement is missing.

Otherwise the Function could be sheduled twice.

**EDIT2**:
The AI.lua builds building replacements without checking for available space. It builds over units, buildings and walls. I did't find a corresponding lua command to fix this. I guess it needs its own Issue.

### PR DESCRIPTION
**Pull request by abcdefg30**
Supersedes #15946.
Makes use of #16277 in TD.

Closes #9456.
Closes #16833.

~~I ported all maps to `nod05` - the rest will follow soon, hopefully.~~

### CODE REVIEW

File: mods/cnc/rules/campaign-maprules.yaml [hunk 1]
```diff
@@ -15,6 +15,13 @@ World:
 	MapOptions:
 		ShortGameCheckboxLocked: True
 		ShortGameCheckboxEnabled: False
+	ScriptLobbyDropdown@difficulty:
+		ID: difficulty
+		Label: Difficulty
+		Values:
+			normal: Normal
+		Default: normal
+		#Locked: true
```
Comment (abcdefg30) [inline mods/cnc/rules/campaign-maprules.yaml]: This will be restored once I ported all missions.

**Comment by fruestueck** [comment]
Could you add ` guard.Stop()` before `guard.Guard(building)` in the `GuardBase` function of mission `nod08a-AI.lua`, `nod08b-AI.lua` and `nod09-AI.lua` when you're at it.

The Orcas guarding the GDIBase will get really confused when you attack two different buildings several times.

**Comment by abcdefg30** [comment]
Updated. This covers all missions now. Most of the AI stuff is still somewhat bogus, but will be replaced anyway once we have settled on an approach in #17129.

**Comment by reaperrr** [comment]
@abcdefg30 Have you tested all missions yourself (or anyone else)?

I'm asking because I don't have the time to do that myself, and the sheer size of the PR plus my own limited lua experience makes a "reading only" review unreliable at best, so the only chance I see to get this merged without splitting it into a series of smaller PRs would be if we could get confirmation from you and/or other testers that the missions indeed still work.

**Comment by abmyii** [comment]
> Have you tested all missions yourself (or anyone else)?

I'd be happy to chip in if there are too many maps for @abcdefg30 to test alone.

**Comment by abcdefg30** [comment]
I did play every mission back in October (latest revision of this PR).
@abmyii playtesting would still be appreciated (just to make sure there are no silly crashes that I overlooked).

**Comment by abmyii** [comment]
Tried the first three missions - all good!

**Comment by abcdefg30** [comment]
Rebased.

**Review by reaperrr — APPROVED** [review]
LGTM

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/254152a1dd36a247e06bd9e509997fb2526314a3...8d65c08878090913d563bff9c5707d0cf64c5a8c

**Pull request merged**